robot/impl/local_robot.go

```diff
@@ -1207,6 +1207,9 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 			} else {
 				canReconfigure, err := r.checkMaintenanceSensorReadings(ctx, newConfig.MaintenanceConfig.MaintenanceAllowedKey, sensorComponent)
 				if !canReconfigure {
+					if err != nil {
+						r.logger.CErrorw(ctx, "error reading maintenance sensor", "error", err)
+					}
 					r.logger.Info("maintenance_allowed_key found from readings on maintenance sensor. Skipping reconfiguration.")
 					diff, err := config.DiffConfigs(*r.Config(), *newConfig, false)
 					if err != nil {
```

Comment:
https://github.com/viamrobotics/rdk/pull/4509/files#diff-0b975da6011656cd3f2caf0435023573bdb5ac7c6a425e7917dd5cc9d6be63e4R1224 case can no longer be true, right? can you remove?

Reply:
yupp thats correct, checkMaintenanceSensorReadings only returns errors when canReconfigure is false. Removing this